### PR TITLE
Hs 352 update profile validation phone input

### PR DESCRIPTION
**Pull request by dlvinh**

### CODE REVIEW

File: lib/features/booking/add_inspection_booking_screen.dart [hunk 1]
```diff
@@ -94,7 +94,14 @@ class _AddInspectionBookingBody extends State<AddInspectionBookingBody> {
                 ),
                 context: context,
                 builder: (_) {
-                  return const UpdatePhoneNoBottomSheetView();
+                  return BlocProvider.value(
+                      value: BlocProvider.of<MyProfileCubit>(context),
```
Comment (suesitran) [inline lib/features/booking/add_inspection_booking_screen.dart:98]: Where is MyProfileCubit created? Can you be sure this cubit is always created at this time?

File: lib/features/booking/add_inspection_booking_screen.dart [hunk 2]
```diff
@@ -94,7 +94,14 @@ class _AddInspectionBookingBody extends State<AddInspectionBookingBody> {
...
+                      child: BlocBuilder<MyProfileCubit, MyProfileState>(
```
Comment (suesitran) [inline lib/features/booking/add_inspection_booking_screen.dart:99]: Why is BlocBuilder used here when there's no UI update based on state change?

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart [hunk 3]
```diff
@@ -67,39 +78,96 @@ class _UpdatePhoneNoBottomSheetBody
...
+                                                  fontSize:
+                                                      FontStyleGuide.fontSize14,
+                                                  fontWeight: FontWeight.w400,
+                                                  color: HSColor.neutral9)))),
+                                  const SizedBox(width: HsDimens.spacing24),
+                                  Expanded(
+                                      child: TextFormField(
+                                    key: const Key('phoneNo'),
+                                    onChanged: (value) {
+                                      setState(() {
```
Comment (suesitran) [inline lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart:107]: Please use bloc, avoid using setState when bloc is in used

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart [hunk 4]
```diff
@@ -112,11 +180,82 @@ class _UpdatePhoneNoBottomSheetBody
...
+  PhoneNumberInputFormatter(this.error);
+  @override
+  TextEditingValue formatEditUpdate(
+      TextEditingValue oldValue, TextEditingValue newValue) {
+    // FORMAT PHONE NUMBER
+    if (oldValue.text.length < newValue.text.length) {
+      if (newValue.text.length == 5) {
+        return TextEditingValue(
+            text:
+                '${oldValue.text.substring(0, 4)} ${newValue.text.substring(newValue.text.length - 1)}');
```
Comment (suesitran) [inline lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart:214]: What format is this?

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart [hunk 5]
```diff
@@ -112,11 +180,82 @@ class _UpdatePhoneNoBottomSheetBody
...
+      }
+
+      if (newValue.text.length == 9) {
+        return TextEditingValue(
+            text:
+                '${oldValue.text.substring(0, 8)} ${newValue.text.substring(newValue.text.length - 1)}');
```
Comment (suesitran) [inline lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart:220]: What format is this?

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart [hunk 6]
```diff
@@ -112,11 +180,82 @@ class _UpdatePhoneNoBottomSheetBody
...
+      error.value = PhoneNumberErrorType.wrongCode;
+      return newValue;
+    }
+    error.value = null;
+    return newValue;
+  }
+}
+
+enum PhoneNumberErrorType {
+  minLength(),
```
Comment (suesitran) [inline lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart:247]: Enum does not need () if there's no param assigned to it.

**Review by suesitran — CHANGES_REQUESTED** [review]

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart (hunk 6 above)
Comment (dlvinh) [reply lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart:247]: fiX

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart (hunk 4 above)
Comment (dlvinh) [reply lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart:214]: The format of input text must be XXXX XXX XXX so I came up with this approach.

File: lib/features/booking/add_inspection_booking_screen.dart (hunk 1 above)
Comment (dlvinh) [reply lib/features/booking/add_inspection_booking_screen.dart]: I already changed cubit that will match with parent cubit. MyProfileCubit will not be used any more, instead, I'm using AddInspectionCubit

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart (hunk 5 above)
Comment (dlvinh) [reply lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart:223]: same answer with other conversation

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart (hunk 3 above)
Comment (dlvinh) [reply lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart]: Fixed with cubit

File: devtools_options.yaml [hunk 7]
```diff
@@ -0,0 +1 @@
+extensions:
```
Comment (suesitran) [inline devtools_options.yaml:1]: what is this change for?

File: lib/features/booking/add_inspection_booking_screen.dart [hunk 8]
```diff
@@ -94,7 +94,9 @@ class _AddInspectionBookingBody extends State<AddInspectionBookingBody> {
                 ),
                 context: context,
                 builder: (_) {
-                  return const UpdatePhoneNoBottomSheetView();
```
Comment (suesitran) [inline lib/features/booking/add_inspection_booking_screen.dart:97]: why do u need to add a `BlocProvider` here? `AddInspectionBookingCubit` is created in the same screen (line 28 above), you can just reuse it here without providing a new cubit

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart [hunk 9]
```diff
@@ -67,39 +78,96 @@ class _UpdatePhoneNoBottomSheetBody
...
-                                  style: const ButtonStyle(
-                                      alignment: Alignment.centerLeft,
-                                      backgroundColor: MaterialStatePropertyAll(
-                                          HSColor.neutral10),
-                                      foregroundColor:
-                                          MaterialStatePropertyAll<Color>(
-                                              HSColor.black),
-                                      textStyle:
-                                          MaterialStatePropertyAll<TextStyle>(
+                          ValueListenableBuilder<PhoneNumberErrorType?>(
```
Comment (suesitran) [inline lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart:81]: what UI will be showed when there's no error in phone number?

File: lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart [hunk 10]
```diff
@@ -112,11 +180,85 @@ class _UpdatePhoneNoBottomSheetBody
...
+                                onPressed: context
+                                            .read<AddInspectionBookingCubit>()
+                                            .phoneNo ==
+                                        null
+                                    ? null
+                                    : value?.phoneNumberError != null
+                                        ? null
+                                        : () {
+                                            // TODO: SAVE AND UPLOAD TO DATABASE
+                                          });
```
Comment (suesitran) [inline lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart:197]: please bring this logic into cubit

**Review by suesitran — CHANGES_REQUESTED** [review]

File: lib/features/booking/add_inspection_booking_screen.dart [hunk 11]
```diff
@@ -97,9 +97,14 @@ class _AddInspectionBookingBody extends State<AddInspectionBookingBody> {
                   return const UpdatePhoneNoBottomSheetView();
                 }).then((value) {
               if (mounted) {
-                context
-                    .read<AddInspectionBookingCubit>()
-                    .validateBookingInspectionButton();
+                if (value != null) {
+                  // value return will be a string of number that entered from bottom sheet
+                  // TODO Handle uploading phone number
```
Comment (suesitran) [inline lib/features/booking/add_inspection_booking_screen.dart:102]: what to upload when value is null?

**Pull request merged**